postgis/gserialized_gist_2d.c

```diff
@@ -1302,6 +1304,96 @@ Datum gserialized_gist_same_2d(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(result);
 }
 
+PG_FUNCTION_INFO_V1(gserialized_gist_sortsupport_2d);
+/*
+ * Hash comparator of `BOX2DF` data type
+ * Notice: the GSERIALIZED GiST uses BOX2DF as index elements
+*/
+static int
+hash_cmp(Datum a, Datum b, SortSupport ssup)
+{
+	if (a > b)
+		return 1;
+	else if (a < b)
+		return -1;
+	else
+		return 0;
+}
+
+static bool
+hash_abbrev_abort(int memtupcount, SortSupport ssup)
+{
+	return false;
+}
+
+/*
+ * `BOX2DF`(geometry) data type converter
+*/
+static Datum
+hash_abbrev_convert(Datum original, SortSupport ssup)
+{
+	BOX2DF *box = (BOX2DF *)original;
+
+	union floatuint {
+		uint32_t u;
+		float f;
+	};
+
+	union floatuint x, y;
+	x.f = (box->xmax + box->xmin) / 2;
+	y.f = (box->ymax + box->ymin) / 2;
+
+	return (Datum)uint32_hilbert(y.u, x.u);
```

Comment:
won't work on 32-bit machines since uint32_hilbert always returns 64-bit value.
you may see here how it was solved before: https://github.com/postgis/postgis/pull/429

Reply:
that is different kind of hash. hilbert is not aimed at uniqueness, but rather at spatial locality.

Reply:
yeah, sorry.
I checked btree and it enables abbrev comparator only for 64bit. I think the same should be done for GIST sortsupport so at least it will work correctly. https://github.com/postgis/postgis/blob/5f4b44edb30c8609d3f812d1f0bc6fd07d750f0f/postgis/lwgeom_btree.c#L189